doc/user_guide/cluster.rst

```diff
@@ -0,0 +1,151 @@
+:ref:`[Lerotic2004] <Lerotic2004>`.
+
+
+Nomenclature
+------------
+
+Taking the example of a 1D Signal of dimensions `(20, 10|4)` containing the
+dataset, we say there are 200 *samples*. The four measured parameters are the
+*features*. If we choose to search for 3 clusters within this dataset, we
+derive two main values: the `labels`, of dimensions `(20, 10|3)` (each
+sample is assigned a label to each cluster), and the `centers`, of
+dimensions `(3, 4)` (each centre has a coordinate in each feature).
+
+
+Example
+-------
+
+We can use the `make_blobs` function supplied by `scikit-learn` to see how
+clustering might work in practice.
+
+.. code-block:: python
+
+    >>> import hyperspy.api as hs
+    >>> from sklearn.datasets import make_blobs
+    >>> data = hs.signals.Signal1D(
+    >>>     make_blobs(
+    >>>         n_samples=200,
+    >>>         n_features=10,
+    >>>         shuffle=False)[0].reshape(20, 10, 10))
+	
+
+The resultant signal contains 3 distinct "types" of signal. 
+If we examine the signal using PCA we can see that there are 3 regions but
+there interpretation of the signal is a little ambigous.  
+
+To see how cluster analysis works it's best to first examine the signal.
+Moving around the image you should be able to see 3 distinct regions in which
+the 1D signal modulates slightly.  
+
+.. code-block:: python
+
+    >>> data.plot()
+
+
+If we then perform PCA we start to see the 3 regions a little more clearly but
+the factors and loadings don't match up with the original 1D signals or image.
+
+.. code-block:: python
+
+    >>> data.decomposition()
+    >>> data.plot_decomposition_results()
+
+
+We can then cluster, using the decomposition results, to find similar regions
+and the representative features in those regions. 
+This indentifies 3 regions and the average or representative 1D signals in 
+those regions
+
+.. code-block:: python
+
+    >>> data.cluster_analysis(3)
+    >>> data.plot_cluster_results()
+
+
+To see what the labels the cluster algorithm has assigned you can inspect
+
+.. code-block:: python
+
+    >>> data.learning_results.cluster_membership
+
+
+This are split into a cluster_labels array to help plotting and masking
+
+.. code-block:: python
+
+    >>> data.learning_results.cluster_labels
+
+
+kmeans and agglomerative clustering methods are currently supported and 
+additional keywords can be passed directly to the scikit learn methods.
+
+.. code-block:: python
+
+    >>> data.cluster_analysis(3, algorithm='agglomerative',
+                              kwargs={affinity='cosine', linkage='average'})
+    >>> data.plot_cluster_results()
+
+
+In this case we know there are 3 signals but for real examples the difficulty
+can be how to define the number of clusters to use.  
+There are a number of metrics for determining the number of clusters. 
```

Comment:
Make a list of the different metrics to make it more clear?

Reply:
updated the doc